src/inmanta/app.py

```diff
@@ -71,14 +71,27 @@
 LOGGER = logging.getLogger("inmanta")
 
 
-@command("server", help_msg="Start the inmanta server")
+def server_parser_config(parser: argparse.ArgumentParser, parent_parsers: abc.Sequence[argparse.ArgumentParser]) -> None:
+    parser.add_argument(
+        "--db-wait-time",
+        default=0,
+        type=int,
+        dest="db_wait_time",
+        help="Maximum time in seconds the server will wait for the database to be up before starting. "
+        "A value of 0 means the server will not wait. If set to a negative value, the server will wait indefinitely.",
+    )
+
+
+@command("server", help_msg="Start the inmanta server", parser_config=server_parser_config)
 def start_server(options: argparse.Namespace) -> None:
     if options.config_file and not os.path.exists(options.config_file):
         LOGGER.warning("Config file %s doesn't exist", options.config_file)
 
     if options.config_dir and not os.path.isdir(options.config_dir):
         LOGGER.warning("Config directory %s doesn't exist", options.config_dir)
 
+    Config.set("database", "wait_time", str(options.db_wait_time))
```

Comment:
I think you have to check if it is set here, otherwise it will always overwrite the config from the file